### DISCUSSION TITLE
Monorepo Support - Adding components to the wrong directory

### DISCUSSION THREAD

**Discussion by bllakcn** [shadcn-ui/ui#6194] [Q&A]
I have been testing the new canary CLI to add components to a Turborepo. I have followed the docs for the installation. I have the default Turborepo configuration, created by the cli `pnpm dlx create-turbo@latest`

```
❯ pnpm dlx shadcn@canary add chart

✔ Checking registry.
✔ Installing components.
✔ Created 2 files:
  - packages/ui/@workspace/ui/components/card.tsx
  - packages/ui/@workspace/ui/components/chart.tsx
  ```

Expected result is
```
  - packages/ui/src/components/card.tsx
  - packages/ui/src/components/chart.tsx
```
I confirm that adding a component to a Turborepo created by `pnpm dlx shadcn@canary init` works fine. I cross checked all the config files but could not find a difference.

The components.json files are as suggested in the [docs](https://ui.shadcn.com/docs/monorepo)
I cannot find what I am missing.

Mentioned here as well #6399

**Comment by jermahs** [comment]
This isn't completely related, but I was finding it frustrating that when installing blocks or components from `apps/web` that it would install the components in `@workspace/ui/src/components` just as I expected it to, but the hooks were in the project itself `@workspace/web/hooks/use-mobile` for instance when installing the `sidebar`

```
$ cd apps/web
$ pnpm dlx shadcn@latest add sidebar
```

This was resolved by `cd` into the `packages/ui` and installing it there instead of in `apps/web`.

I think I could simply change `components.json` in `apps/web` and change the `hooks` alias from `@/hooks` to `@spigot/ui/hooks`, but I'm not sure what else that would affect.

**Reply by jermahs** [reply]
FWIW and a bit unrelated, the `@radix-ui/*` dependencies are not being installed correct and I often have to run the appropriate `pnpm add` command for the dependencies such as the following for sidebar. `pnpm add @radix-ui/react-collapsible`
